src/stcal/ramp_fitting/likely_fit.py

```diff
@@ -0,0 +1,1335 @@
+        # in the ramp.  If there are only two left we have no way of
+        # choosing which one is "good".  If there are three left we
+        # run into trouble in case we need to discard two.
+        recheck[np.sum(diffs2use, axis=0) <= 3] = False
+
+    return diffs2use, countrate
+
+
+def get_readtimes(ramp_data):
+    """
+    Get the read times needed to compute the covariance matrices.  If there is
+    already a read_pattern in the ramp_data class, then just get it.  If not, then
+    one needs to be constructed.  If one needs to be constructed it is assumed the
+    groups are evenly spaced in time, as are the frames that make up the group.  If
+    each group has only one frame and no group gap, then a list of the group times
+    is returned.  If nframes > 0, then a list of lists of each frame time in each
+    group is returned with the assumption:
+        group_time = (nframes + groupgap) * frame_time
+
+    Parameters
+    ----------
+    ramp_data : RampData
+        Input data necessary for computing ramp fitting.
+
+    Returns
+    -------
+    readtimes : list
+        A list of frame times for each frame used in the computation of the ramp.
+    """
+    if ramp_data.read_pattern is not None:
+        return ramp_data.read_pattern
+
+    ngroups = ramp_data.data.shape[1]
+    # rtimes = [(k + 1) * ramp_data.group_time for k in range(ngroups)]   # XXX Old
+    tot_frames = ramp_data.nframes + ramp_data.groupgap
+    tot_nreads = np.arange(1, ramp_data.nframes + 1)
+    rtimes = [
+        (tot_nreads + k * tot_frames) * ramp_data.frame_time for k in range(ngroups)
+    ]
+
+    return rtimes
+
+
+def compute_image_info(integ_class, ramp_data):
+    """
+    Compute the diffs2use mask based on DQ flags of a row.
+
+    Parameters
+    ----------
+    integ_class : IntegInfo
+        Contains the rateints product calculations.
+
+    ramp_data : RampData
+        Input data necessary for computing ramp fitting.
+
+    Returns
+    -------
+    image_info : tuple
+        The list of arrays for the rate product.
+    """
+    if integ_class.data.shape[0] == 1:
+        data = integ_class.data[0, :, :]
+        dq = integ_class.dq[0, :, :]
+        var_p = integ_class.var_poisson[0, :, :]
+        var_r = integ_class.var_rnoise[0, :, :]
+        var_e = integ_class.err[0, :, :]
+        return (data, dq, var_p, var_r, var_e)
+
+    dq = utils.dq_compress_final(integ_class.dq, ramp_data)
+
+    inv_err2 = 1.0 / (integ_class.err**2)
+    weight = inv_err2 / inv_err2.sum(axis=0)
+    weight2 = weight**2
+
+    err2 = np.sum(integ_class.err**2 * weight2, axis=0)
+
+    err = np.sqrt(err2)
+    var_p = np.sum(integ_class.var_poisson * weight2, axis=0)
+    var_r = np.sum(integ_class.var_rnoise * weight2, axis=0)
+    slope = np.sum(integ_class.data * weight, axis=0)
+
+    return (slope, dq, var_p, var_r, err)
+
+
+def determine_diffs2use(ramp_data, integ, row, diffs):
```

Comment:
Do we want to add an optional argument for whether to propagate the jump flags?

Reply:
That happens here: https://github.com/kmacdonald-stsci/stcal/blob/b92cf4ebef7422f7127e9cb87b51770725d05690/src/stcal/ramp_fitting/likely_fit.py#L108